packages/evm/src/opcodes/codes.ts

```diff
@@ -52,159 +52,173 @@ export type OpcodeList = Map<number, Opcode>
-  0x03: { name: 'SUB', isAsync: false, dynamicGas: false },
-  0x04: { name: 'DIV', isAsync: false, dynamicGas: false },
-  0x05: { name: 'SDIV', isAsync: false, dynamicGas: false },
-  0x06: { name: 'MOD', isAsync: false, dynamicGas: false },
-  0x07: { name: 'SMOD', isAsync: false, dynamicGas: false },
-  0x08: { name: 'ADDMOD', isAsync: false, dynamicGas: false },
-  0x09: { name: 'MULMOD', isAsync: false, dynamicGas: false },
-  0x0a: { name: 'EXP', isAsync: false, dynamicGas: true },
-  0x0b: { name: 'SIGNEXTEND', isAsync: false, dynamicGas: false },
+  0x00: defaultOp('STOP'),
+  0x01: defaultOp('ADD'),
+  0x02: defaultOp('MUL'),
+  0x03: defaultOp('SUB'),
+  0x04: defaultOp('DIV'),
+  0x05: defaultOp('SDIV'),
+  0x06: defaultOp('MOD'),
+  0x07: defaultOp('SMOD'),
+  0x08: defaultOp('ADDMOD'),
+  0x09: defaultOp('MULMOD'),
+  0x0a: dynamicGasOp('EXP'),
+  0x0b: defaultOp('SIGNEXTEND'),
 
   // 0x10 range - bit ops
-  0x10: { name: 'LT', isAsync: false, dynamicGas: false },
-  0x11: { name: 'GT', isAsync: false, dynamicGas: false },
-  0x12: { name: 'SLT', isAsync: false, dynamicGas: false },
-  0x13: { name: 'SGT', isAsync: false, dynamicGas: false },
-  0x14: { name: 'EQ', isAsync: false, dynamicGas: false },
-  0x15: { name: 'ISZERO', isAsync: false, dynamicGas: false },
-  0x16: { name: 'AND', isAsync: false, dynamicGas: false },
-  0x17: { name: 'OR', isAsync: false, dynamicGas: false },
-  0x18: { name: 'XOR', isAsync: false, dynamicGas: false },
-  0x19: { name: 'NOT', isAsync: false, dynamicGas: false },
-  0x1a: { name: 'BYTE', isAsync: false, dynamicGas: false },
+  0x10: defaultOp('LT'),
+  0x11: defaultOp('GT'),
+  0x12: defaultOp('SLT'),
+  0x13: defaultOp('SGT'),
+  0x14: defaultOp('EQ'),
+  0x15: defaultOp('ISZERO'),
+  0x16: defaultOp('AND'),
+  0x17: defaultOp('OR'),
+  0x18: defaultOp('XOR'),
+  0x19: defaultOp('NOT'),
+  0x1a: defaultOp('BYTE'),
 
   // 0x20 range - crypto
-  0x20: { name: 'KECCAK256', isAsync: false, dynamicGas: true },
+  0x20: dynamicGasOp('KECCAK256'),
 
   // 0x30 range - closure state
-  0x30: { name: 'ADDRESS', isAsync: true, dynamicGas: false },
-  0x31: { name: 'BALANCE', isAsync: true, dynamicGas: true },
-  0x32: { name: 'ORIGIN', isAsync: true, dynamicGas: false },
-  0x33: { name: 'CALLER', isAsync: true, dynamicGas: false },
-  0x34: { name: 'CALLVALUE', isAsync: true, dynamicGas: false },
-  0x35: { name: 'CALLDATALOAD', isAsync: true, dynamicGas: false },
-  0x36: { name: 'CALLDATASIZE', isAsync: true, dynamicGas: false },
-  0x37: { name: 'CALLDATACOPY', isAsync: true, dynamicGas: true },
-  0x38: { name: 'CODESIZE', isAsync: false, dynamicGas: false },
-  0x39: { name: 'CODECOPY', isAsync: false, dynamicGas: true },
-  0x3a: { name: 'GASPRICE', isAsync: false, dynamicGas: false },
-  0x3b: { name: 'EXTCODESIZE', isAsync: true, dynamicGas: true },
-  0x3c: { name: 'EXTCODECOPY', isAsync: true, dynamicGas: true },
+  0x30: asyncOp('ADDRESS'),
```

Comment:
Wow, just noticing this here, it seem that we have marked some opcodes (also in the original) with the wrong type. `ADDRESS` is definitely not async! I will address typing those right in a new PR.

Reply:
Yeah, so I *do* still thing that this new more compact notation is useful (beyond small size gains), since it more points the properties out (I would assume that you actually spotted due to this new form factor).

Reply:
Yup, that's true. I actually thought you converted the opcodes wrong until I realized that it was wrong in the first place :joy: This is definitely better and makes it much more readable! :smile:

Reply:
Side question (out of interest): does this have any effect (on performance e.g.) if changed back to `sync`? Or is this more for correctness?

Reply:
I think it will have a tiny impact since if it is async, it will likely end up in a promise queue, so this would at least "save" a push and pop operation (but this depends on the inner workings of node, so I have no idea if this is actually the case). But it should be changed nevertheless for correctness.